tests/PHPStan/Analyser/nsrt/in_array_loose.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace InArrayLoose;
+
+use function PHPStan\Testing\assertType;
+
+class Foo
+{
+	public function looseComparison(
+		string $string,
+		int $int,
+		float $float,
+		bool $bool,
+		string|int $stringOrInt,
+		string|null $stringOrNull,
+	): void {
+		if (in_array($string, ['1', 'a'])) {
+			assertType("'1'|'a'", $string);
+		}
+		if (in_array($string, [1, 'a'])) {
+			assertType("string", $string);
```

Comment:
Some assertion will be improved later for example when https://phpstan.org/r/49e37d0b-b17a-4a90-9fff-0618a5c7fc41 will be improved.